description.md

```diff
@@ -0,0 +1,937 @@
+```
+
+In addition, the `takerInteraction` will be:
+
+```bash
+# takerInteraction calldata
+# target address
+0165878A 594ca255 338adfa4 d48449f6 9242Eb8F
+# extraData passed to the call
+00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000003
+```
+
+# Filling an order
+
+## How to fill an order
+
+To fill an order, a taker should call one of the functions defined in the `IOrderMixin` interface and supply the order along with any extensions, the filling amount, a signature, and taker filling options. The functions defined by the interface vary, depending on: whether the order is filled with or without extensions, whether the taker is the fund receiver or not, and whether the order is filled with the taker's permit or not. 
+
+The full list is as follows:
+
+- `fillOrder` - fills a simple order without extensions.
+- `fillOrderExt` - allows the specification of extensions used for the order.
+- `fillOrderTo` - allows the specification of the maker's funds destination instead of `msg.sender`.
+- `fillOrderToExt` - allows the specification of the maker's funds destination and extensions used for the order.
+- `fillOrderToWithPermit` - allows the specification of the maker's funds destination and calls permit before filling the order.
+- `fillContractOrder` - uses contract-based signatures.
+- `fillContractOrderWithPermit` - uses contract-based signatures and the taker's permit.
+- `fillContractOrderExt` - uses contract-based signatures, the taker's permit, and allows the specification of the order's extensions.
+
+All the functions have a similar signature. Here are examples showing the possible arguments and their descriptions:
+
+```solidity
+// Fill from EOA
+function fillOrderToExt(
+        IOrderMixin.Order calldata order,
+        bytes32 r,
+        bytes32 vs,
+        uint256 amount,
+        TakerTraits takerTraits,
+        address target,
+        bytes calldata interaction,
+        bytes calldata extension
+    ) public payable returns(uint256 makingAmount, uint256 takingAmount, bytes32 orderHash) {
+
+// Fill from smart contract
+function fillContractOrderExt(
+        IOrderMixin.Order calldata order,
+        bytes calldata signature,
+        uint256 amount,
+        TakerTraits takerTraits,
+        address target,
+        bytes calldata interaction,
+        bytes calldata permit,
+        bytes calldata extension
+    ) public returns(uint256 makingAmount, uint256 takingAmount, bytes32 orderHash)
+```
+
+| Argument | Type | Description |
+| --- | --- | --- |
+| order | `Order` (calldata) | The order structure to be filled.  |
+| r | `bytes32` | The r-component of the maker’s signature to check that the order hash is signed by the maker. |
+| vs | `bytes32` | The vs-component of the maker’s signature to check that the order hash is signed by the maker. |
+| signature | `bytes calldata` | The signature used to verify the order. It is used for contract-signed orders only. |
```

Comment:
It could be useful to mention ERC-1271 and `isValidSignature` method

Reply:
fixed